### PR TITLE
Implementation completed

### PR DESCRIPTION
**Pull request by houman-sanati**

### CODE REVIEW

File: src/app/utils/Strings.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+enum Strings {
```
Comment (hadideveloper) [inline src/app/utils/Strings.ts:1]: These messages need improvement.

File: src/app/components/passphrase-dialog/passphrase-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  passphraseForm: FormGroup
+  constructor(
+    private formBuilder: FormBuilder,
+    public dialogRef: MatDialogRef<PassphraseDialogComponent>
+  ) { }
+
+  ngOnInit(): void {
+    this.passphraseForm = this.formBuilder.group({
+      passphrase: ['', Validators.required],
+      confirmPassphrase: ['', Validators.required],
```
Comment (hadideveloper) [inline src/app/components/passphrase-dialog/passphrase-dialog.component.ts:20]: Where do you check if confirmPassphrase equals to passphrase?

**Review by hadideveloper — CHANGES_REQUESTED** [review]

File: package-lock.json [hunk 3]
```diff
@@ -0,0 +1,19760 @@
+{
```
[the remark this answers is not in the document]
Comment (houman-sanati) [reply package-lock.json]: fixed in [d9cd079](https://github.com/prokey-io/prokey-link/pull/1/commits/d9cd0799e6a474353017b8e2d40220c6ee0fbc9c)

File: src/app/app.component.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+import CommandType from './models/CommandType'
+import { MessageDialogComponent } from './components/dialog/dialog.component';
+import Strings from './utils/Strings';
+import { getDialogSettings, openDialog } from './utils/dialogUtil';
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss']
+})
+export class AppComponent implements OnInit {
```
[the remark this answers is not in the document]
Comment (houman-sanati) [reply src/app/app.component.ts:22]: fixed in [535863e](https://github.com/prokey-io/prokey-link/pull/1/commits/535863e63302b6c77eee0e7b71d57f00252c5844)

File: src/app/app.component.ts [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+  device: Device = null
```
[the remark this answers is not in the document]
Comment (houman-sanati) [reply src/app/app.component.ts]: fixed in [66ca249](https://github.com/prokey-io/prokey-link/pull/1/commits/66ca249d7fc4af165757c7503c926725d4311c72)

File: src/app/app.component.ts [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+  closeDeviceActionDialog() {
+    this.deviceActionDialog?.close()
+  }
+
+  setListeners() {
+    this.device.AddOnFailureCallBack(async (reason) => {
+      this.closeDeviceActionDialog()
+      this.isDeviceConnected = false
+      switch (reason.code) {
+        case 4:
```
[the remark this answers is not in the document]
Comment (houman-sanati) [reply src/app/app.component.ts]: fixed in [8a34c35](https://github.com/prokey-io/prokey-link/pull/1/commits/8a34c35b2a4f27df0f81380e600708b1ece6ce6c)

File: src/app/components/passphrase-dialog/passphrase-dialog.component.ts (hunk 2 above)
Comment (houman-sanati) [reply src/app/components/passphrase-dialog/passphrase-dialog.component.ts]: fixed in [a2f53c8](https://github.com/prokey-io/prokey-link/pull/1/commits/a2f53c8a87dc0acaa82c0609f5d3a35c031997f1)

File: src/app/utils/Strings.ts (hunk 1 above)
Comment (houman-sanati) [reply src/app/utils/Strings.ts:1]: fixed in [ce84bd9](https://github.com/prokey-io/prokey-link/pull/1/commits/ce84bd98c805870fccd5ae7247d352ef41c161dc)

**Review by hadideveloper — APPROVED** [review]

**Pull request merged**